addons/stock_account/models/product.py

```diff
@@ -8,17 +8,55 @@
 from collections import defaultdict
 from datetime import datetime
 
+from odoo.addons.stock_account import const
+
 
 class ProductTemplate(models.Model):
     _inherit = ['product.template']
 
-    cost_method = fields.Selection(related="categ_id.property_cost_method", readonly=True)
-    valuation = fields.Selection(related="categ_id.property_valuation", readonly=True)
+    cost_method = fields.Selection(
+        string="Cost Method",
+        selection=const.COST_METHOD,
+        compute='_compute_cost_method',
+    )
+    valuation = fields.Selection(
```

Comment:
Since `valuation` and `cost_method` are now non-stored computed fields, maybe we shouldn't try to set a value directly to them in the following tests anymore:
https://github.com/odoo/odoo/blob/c551b9e850bebb33a6d434cf6538e182c60d4db0/addons/purchase_stock/tests/test_stockvaluation.py#L1081-L1082
https://github.com/odoo/odoo/blob/c551b9e850bebb33a6d434cf6538e182c60d4db0/addons/stock_account/tests/test_stockvaluation.py#L3824
https://github.com/odoo/odoo/blob/c551b9e850bebb33a6d434cf6538e182c60d4db0/addons/stock_dropshipping/tests/test_purchase_order.py#L36